Lib/test/test_strftime.py

```diff
@@ -54,12 +55,17 @@ def _update_variables(self, now):
         self.now = now
 
     def setUp(self):
+        self.saved_locale = None
         try:
             import java
             java.util.Locale.setDefault(java.util.Locale.US)
         except ImportError:
-            import locale
-            locale.setlocale(locale.LC_TIME, 'C')
+            self.saved_locale = setlocale(LC_TIME)
```

Comment:
I suggest to add here:

self.addCleanup(setlocale, LC_TIME, self.saved_locale)

With that, you avoid tearDown() (which is not called if the test is interrupted by CTRL+c if I recall correctly), and you may even use a local variable instead of an attribute.

Reply:
Yes, much better. Thanks for your reviews Victor.